core/engine/src/object/mod.rs

```diff
@@ -464,6 +465,17 @@ pub struct FunctionBinding {
     pub(crate) name: JsString,
 }
 
+impl From<JsStr<'_>> for FunctionBinding {
+    #[inline]
+    fn from(name: JsStr<'_>) -> Self {
+        let name: JsString = name.into();
+        Self {
+            binding: name.clone().into(),
+            name: name.clone(),
+        }
+    }
+}
+
```

Comment:
I think for things that require converting to `JsString` we should require the `JsString` itself, instead of a `JsStr`. This mostly matches Rust's APIs around collections, which commonly take the inserted key by value instead of by reference.

This probably applies to this, set methods on objects, and in general any method that necessarily needs to take a `JsString` by value.

TLDR: We should remove implicit conversions like this, in favour of taking `JsString` by value.

Reply:
I do see the utility of `JsStr`, but that will be more helpful to avoid allocations on all the getter methods of the engine.